File: internal/model/category.go
package model

import (
	"time"
)

// Category 分类模型
type Category struct {
	ID        uint      `gorm:"primaryKey"`
	UserID    uint      `gorm:"index;not null"`                 // 所属用户 ID
	Name      string    `gorm:"size:100;not null"`              // 分类名称
	Color     string    `gorm:"size:20;default:'#000000'"`       // 分类颜色
	Icon      string    `gorm:"size:50"`                        // 分类图标
	SortOrder int       `gorm:"default:0"`                      // 排序
	CreatedAt time.Time
	UpdatedAt time.Time
}

// TableName 指定表名
func (Category) TableName() string {
	return "categories"
}

File: src/model/category.ts
import { sqliteTable, text, integer, index } from "drizzle-orm/sqlite-core";

export const categories = sqliteTable(
  "categories",
  {
    id: integer("id").primaryKey({ autoIncrement: true }),
    userId: integer("user_id").notNull(),
    name: text("name", { length: 100 }).notNull(),
    color: text("color", { length: 20 }),
    icon: text("icon", { length: 50 }),
    sortOrder: integer("sort_order").notNull().default(0),
    createdAt: text("created_at").notNull(),
    updatedAt: text("updated_at").notNull(),
  },
  (table) => ({
    userIdIdx: index("idx_categories_user_id").on(table.userId),
    nameIdx: index("idx_categories_name").on(table.name),
  }),
);

export type Category = typeof categories.$inferSelect;
export type InsertCategory = typeof categories.$inferInsert;
